src/contracts/core/StrategyManager.sol

```diff
@@ -371,6 +355,16 @@ contract StrategyManager is
                     "StrategyManager.queueWithdrawal: cannot queue a withdrawal including Beacon Chain ETH and other tokens");
                 require(shares[i] % GWEI_TO_WEI == 0,
                     "StrategyManager.queueWithdrawal: cannot queue a withdrawal of Beacon Chain ETH for an non-whole amount of gwei");
+                
+                /**
+                 * This decrements the withdrawablRestakedExecutionLayerGwei which is incremented only when a podOwner proves a full withdrawal.
+                 * Remember that withdrawablRestakedExecutionLayerGwei tracks the currently withdrawable ETH from the EigenPod.  
+                 * By doing this, we ensure that the number of shares in EigenLayer matches the amount of withdrawable ETH in 
+                 * the pod plus any ETH still staked on the beacon chain via other validators pointed to the pod. A result of this
+                 * is that this effectively requires a full withdrawal of a validator to queue a withdrawal of beacon chain ETH shares - otherwise
+                 * withdrawablRestakedExecutionLayerGwei is 0. 
```

Comment:
I found this last sentence to be confusingly worded. Perhaps `As a result, a validator must complete a full withdrawal from the execution layer prior to queuing a withdrawal of 'beacon chain ETH shares' via EigenLayer, since otherwise withdrawableRestakedExecutionLayerGwei will be 0`

Reply:
addressed